sqlx-core/src/sqlite/options/connect.rs

```diff
@@ -18,14 +18,23 @@ impl ConnectOptions for SqliteConnectOptions {
             let mut conn = establish(self).await?;
 
             // send an initial sql statement comprised of options
-            //
+
             // page_size must be set before any other action on the database.
-            //
+            let page_size_opt = format!("PRAGMA key = {};", self.page_size);
+            conn.execute(&*page_size_opt).await?;
+
+            // apply custom initial_options
+            match self.initial_options {
+                Some(stmt) => {
+                    conn.execute(&*stmt).await?;
+                }
+                None => (),
+            }
```

Comment:
Unless you intended to add something to the `None` branch:

```suggestion
            if let Some(stmt) = self.initial_options {
                    conn.execute(&*stmt).await?;
            }
```